### DISCUSSION TITLE
Functionality to bulk update users Required action instead of looping with admin endpoint?

### DISCUSSION THREAD

**Discussion by juyoungkimthedev** [keycloak/keycloak#35374] [Q&A]
Do we have a way to update multiple users' Required Actions?
Currently I don't see an admin endpoint to bulk update users so we'd need to use loops to get the users and update the individual users required action using user update admin REST endpoint. Meaning we'd have to call Keycloak's admin endpoint for multiple users which seems inefficient.

It would be useful if we had bulk update endpoint or functionality to update multiple users making less calls to Keycloak's admin endpoint.

Is this possible and can you please recommend efficient ways of achieving this for multiple users?

**Comment by aemaem** [comment]
Apparently, this should be possible with **Keycloak 27**, when Workflows will be officially supported. See the feature description for more information: https://github.com/keycloak/keycloak/issues/39888
